core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/source/mysql/MysqlSource.java

```diff
@@ -0,0 +1,246 @@
+        } catch (SQLException | InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            throw new DataflowException("MysqlSink failed to connect to mysql database." + e.getMessage());
+        }
+    }
+    
+    @Override
+    public Tuple getNextTuple() throws TexeraException {
+        if (cursor == CLOSED) {
+            throw new DataflowException(ErrorMessages.OPERATOR_NOT_OPENED);
+        }
+        try {
+            if(start){
+                System.out.print(generateSqlQuery(predicate));
+                PreparedStatement ps = this.connection.prepareStatement(generateSqlQuery(predicate));
+                /*if(!predicate.getColumn().equals("") && !predicate.getKeywords().equals("")){
+                    ps.setObject(1, predicate.getColumn(), Types.VARCHAR);
+                    ps.setObject(2, predicate.getKeywords(), Types.VARCHAR);
+                    if(predicate.getLimit()!=Integer.MAX_VALUE){
+                        ps.setObject(3, predicate.getLimit(), Types.INTEGER);
+                    }
+                }
+                else{*/
+                if(predicate.getLimit()!=Integer.MAX_VALUE){
+                    ps.setObject(1, predicate.getLimit(), Types.INTEGER);
+                }
+                this.rs = ps.executeQuery();
+                start = false;}
+            //}
+			while (rs.next()) {
+			    List<IField> tb = new ArrayList();
+			    for(Attribute a: this.outputSchema.getAttributes()){
+			        if (a.getType() == AttributeType.STRING){
+			            String value = rs.getString(a.getName());
+			            value = value ==null? "":value;
+			            tb.add(new StringField(value));
+                    }else if (a.getType() == AttributeType.INTEGER){
+                        String value = rs.getString(a.getName());
+                        value = value ==null? "0":value;
+                        tb.add(new IntegerField(Integer.valueOf(value)));
+                    }else if (a.getType() == AttributeType.DOUBLE){
+                        String value = rs.getString(a.getName());
+                        value = value ==null? "0.0":value;
+                        tb.add(new DoubleField(Double.valueOf(value)));
+                    }else if (a.getType() == AttributeType.INTEGER){
+                        String value = rs.getString(a.getName());
+                        value = value ==null? "0":value;
+                        tb.add(new IntegerField(Integer.valueOf(value)));
+                    }else if (a.getType() == AttributeType.DATE){
+                        Date value = rs.getDate(a.getName());
+                        if (value == null) {
+                            SimpleDateFormat fmt = new SimpleDateFormat("yyyy-MM-dd");
+                            try{
+                                value = fmt.parse("0000-00-00");
+                            }catch (ParseException e){
+                                System.out.println("parse error");
```

Comment:
throw the exception, don't catch it and do nothing. If something wrong happens, the user will not know about it.